algorithms/gr-dfsrh/java/DepthFirstSearch.java

```diff
@@ -0,0 +1,73 @@
+import java.io.*; 
+import java.util.*; 
+  
+// This class represents a directed graph using adjacency list 
+// representation 
+class Graph 
+{ 
+    private int V;   // No. of vertices 
+  
+    // Array  of lists for Adjacency List Representation 
+    private LinkedList<Integer> adj[]; 
+  
+    // Constructor 
+    @SuppressWarnings("unchecked") 
+    Graph(int v) 
+    { 
+        V = v; 
+        adj = new LinkedList[v]; 
+        for (int i=0; i<v; ++i) 
+            adj[i] = new LinkedList(); 
+    } 
+  
+    //Function to add an edge into the graph 
+    void addEdge(int v, int w) 
+    { 
+        adj[v].add(w);  // Add w to v's list. 
+    } 
+  
+    // A function used by DFS 
+    void DFSUtil(int v,boolean visited[]) 
+    { 
+        // Mark the current node as visited and print it 
+        visited[v] = true; 
+        System.out.print(v+" "); 
+  
+        // Recur for all the vertices adjacent to this vertex 
+        Iterator<Integer> i = adj[v].listIterator(); 
+        while (i.hasNext()) 
+        { 
+            int n = i.next(); 
+            if (!visited[n]) 
+                DFSUtil(n, visited); 
+        } 
+    } 
+  
+    // The function to do DFS traversal. It uses recursive DFSUtil() 
+    void DFS(int v) 
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6327382)